rand_core/src/impls.rs

```diff
@@ -63,24 +63,46 @@ macro_rules! try_fill_via {
             };
             left.copy_from_slice(&chunk[..n]);
         }
-        Ok(())
     }}
 }
 
-/// Implement `try_fill` via `next_u32`, little-endian order.
-pub fn try_fill_via_u32<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) -> Result<()> {
-    try_fill_via!(rng, next_u32, 4, dest)
+/// Implement `fill_bytes` via `next_u32`, little-endian order.
+pub fn fill_bytes_via_u32<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) {
+    fill_bytes_via!(rng, next_u32, 4, dest)
 }
 
-/// Implement `try_fill` via `next_u64`, little-endian order.
-pub fn try_fill_via_u64<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) -> Result<()> {
-    try_fill_via!(rng, next_u64, 8, dest)
+/// Implement `fill_bytes` via `next_u64`, little-endian order.
+pub fn fill_bytes_via_u64<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) {
+    fill_bytes_via!(rng, next_u64, 8, dest)
 }
 
-/// Implement `try_fill` via `next_u128`, little-endian order.
+/// Implement `fill_bytes` via `next_u128`, little-endian order.
 #[cfg(feature = "i128_support")]
-pub fn try_fill_via_u128<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) -> Result<()> {
-    try_fill_via!(rng, next_u128, 16, dest)
+pub fn fill_bytes_via_u128<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) {
+    fill_bytes_via!(rng, next_u128, 16, dest)
+}
+
+macro_rules! impl_uint_from_fill {
+    ($self:expr, $ty:ty, $N:expr) => ({
+        // Transmute and convert from LE (i.e. byte-swap on BE)
+        debug_assert!($N == ::core::mem::size_of::<$ty>());
+        let mut buf = [0u8; $N];
+        $self.fill_bytes(&mut buf);
```

Comment:
Should *probably* be `buf.to_le()`, but maybe best double-check (bit tired right now). Of course on LE it won't do anything anyway.